### DISCUSSION TITLE
Clarification Needed on Upgrade Script Version for Air-Gap Environments

### DISCUSSION THREAD

**Discussion by Miaoxiang-philips** [k3s-io/k3s#10583] [Q&A] [answered]
Hello,

I am following the [documentation](https://docs.k3s.io/installation/airgap#install-script-method) for upgrading K3s in an air-gap environment. The instructions suggest downloading the latest installation script from https://get.k3s.io because the script may have changed since the last release.

My concern is that the script available at https://get.k3s.io is always the latest version. This raises a question about compatibility with specific K3s versions. For example, if I am upgrading from an older version of K3s to a version that is not the most recent, will the latest script from https://get.k3s.io still be compatible? Is there a way to access the script that matches the K3s version I am upgrading to?

Any clarification on this would be greatly appreciated. Thank you!

**Comment by dereknola** [comment, marked as answer]
The latest install script will be able to install any version of K3s. The install script itself is considered separate from the K3s versions.

There is no reason to you an older version of the install script, its not assigned to any particular release. It would just be a worse version of the install script.
